src/Workspaces/Core/Portable/Execution/Asset.cs

```diff
@@ -121,7 +121,7 @@ internal sealed class AnalyzerReferenceAsset : Asset
         private readonly AnalyzerReference _reference;
 
         public AnalyzerReferenceAsset(Serializer serializer, AnalyzerReference reference, Checksum checksum, string kind) :
-            base(checksum, WellKnownChecksumObjects.AnalyzerReference)
+            base(Checksum.Create(kind, checksum), WellKnownChecksumObjects.AnalyzerReference)
         {
             Contract.Requires(kind == WellKnownChecksumObjects.AnalyzerReference);
```

Comment:
well, I could move the check one level up

http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Execution/AssetBuilder.cs,135

anyway, I am not changing those now. when solution state refactoring will change all these anyway.